packages/cms-lib/themes.js

```diff
@@ -0,0 +1,178 @@
+const fs = require('fs-extra');
+const path = require('path');
+const os = require('os');
+const { promisify } = require('util');
+
+const request = require('request-promise-native');
+const extract = promisify(require('extract-zip'));
+
+const { logger } = require('./logger');
+const {
+  logApiErrorInstance,
+  logFileSystemErrorInstance,
+  logErrorInstance,
+} = require('./errorHandlers');
+
+// TODO: When the boilerplate project cuts a release, use latest instead.
+// https://developer.github.com/v3/repos/releases/#get-the-latest-release
+// https://api.github.com/repos/HubSpot/cms-theme-boilerplate/releases/latest
+const THEME_BOILERPLATE_ZIP_URI =
+  'https://github.com/HubSpot/cms-theme-boilerplate/archive/master.zip';
+const ZIP_CONTENT_TYPE = 'application/zip';
+const TMP_BOILERPLATE_FOLDER_PREFIX = 'hubspot-cms-theme-boilerplate-';
+
+/**
+ * https://developer.github.com/v3/repos/releases/#get-the-latest-release
+ * @param {String|Null} tag - Git tag to fetch for. If omitted latest will be fetched.
+ */
+async function fetchReleaseData(tag = null) {
+  const URI = tag
+    ? `https://api.github.com/repos/HubSpot/cms-theme-boilerplate/releases/tags/${tag}`
+    : 'https://api.github.com/repos/HubSpot/cms-theme-boilerplate/releases/latest';
+  try {
+    return await request.get(URI);
+  } catch (err) {
+    logger.error(`Failed fetching release data for ${tag || 'latest'} theme.`);
+    logApiErrorInstance(err, {
+      request: URI,
+    });
+  }
+  return null;
+}
+
+/**
+ * @param {String} tag - Git tag to fetch for. If omitted latest will be fetched.
+ * @returns {Buffer|Null} Zip data buffer
+ */
+async function downloadCmsThemeBoilerplate(/*tag = null*/) {
+  try {
+    // const releaseData = fetchReleaseData(tag);
+    // if (!releaseData) return;
+    // const { zipball_url: zipUrl, tag_name: tagName } = releaseData;
+    // logger.log(`Fetching theme ${tagName}...`);
+    // TODO: Use `zipUrl` when we start releases.
```

Comment:
@gcorne @axellabs When could we cut a release of the boilerplate so this code can be used? There is only a draft release presently.
https://github.com/HubSpot/cms-theme-boilerplate/releases

Reply:
I'm on it right now!

Reply:
The release is out

Reply:
Thanks @axellabs 
https://api.github.com/repos/HubSpot/cms-theme-boilerplate/releases/latest